File: 卡码/代码随想录/1/08-贪心算法/Js/03-困难/LeetCode_0134.js
//134. 加油站
const canCompleteCircuit = function(gas, cost) {
	let start=0,current=0,total=0;
	for(let i=0;i<gas.length;i++){
		current+=(gas[i]-cost[i]);
		total+=(gas[i]-cost[i]);
		if(current<0){
			current=0;
			start=i+1;
		}
	}
	if(total<0) return -1;
	return start;
};

File: 卡码/代码随想录/1/08-贪心算法/Python/03-困难/LeetCode_0134.py
#134. 加油站
from typing import *

class Solution:
	def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
		start,current,total=0,0,0
		for i in range(len(gas)):
			current+=(gas[i]-cost[i])
			total+=(gas[i]-cost[i])
			if current<0:
				current=0
				start=i+1
		if total<0:
			return -1
		return start